src/System.CommandLine/Argument{T}.cs

```diff
@@ -160,5 +162,36 @@ public void AcceptLegalFileNamesOnly()
                 }
             });
         }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL3050", Justification = "https://github.com/dotnet/command-line-api/issues/1638")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091", Justification = "https://github.com/dotnet/command-line-api/issues/1638")]
+        internal static T? CreateDefaultValue()
+        {
+            if (default(T) is null && typeof(T) != typeof(string))
+            {
+                if (typeof(T).IsArray)
+                {
+                    return (T?)(object)Array.CreateInstance(typeof(T).GetElementType()!, 0);
```

Comment:
We should be able to use https://github.com/dotnet/runtime/issues/76478 once it is implemented in .NET 8. That will allow us to do this without a warning.